### ISSUE TITLE
Training from scratch custom data

### ISSUE THREAD

**Issue by jordigc2** [zxcqlf/MonoViT#10]
Hello Zhaocq,

First of all thank you very much for sharing your amazing work. I managed to integrate the model into the Monodepth training pipeline and train the model using the pre-trained weights that you provided. Even though, the results obtained were not great as with far object/scenes the model was not managing to keep a smooth disparity or properly interpreting the scene. The top-right image corresponds to the ZED2 output depth.
![image](https://user-images.githubusercontent.com/22197158/224310395-a36593f7-fd7c-4431-8185-094ea09b5435.png)

My goal is to be able to train it from "scratch" (using the pre-trained models with ImageNet) as I would like to include a semi-supervised term into the loss function using GT Depth (LIDAR or ZED2 depth)  information to be able to get metric depth and pose and keep consistency with far elements of the image. Inspired by this paper https://arxiv.org/pdf/1910.01765.pdf 

Here I attach some sample images that I am using for training:

![image](https://user-images.githubusercontent.com/22197158/224307297-731e1c08-75d5-4bae-bd0b-dc2ce535c8b8.png) ![image](https://user-images.githubusercontent.com/22197158/224308038-c7a64118-27d1-4a37-ac5e-e447685d4431.png) 
![image](https://user-images.githubusercontent.com/22197158/224308142-6f9561b0-5a06-47cb-b9ff-96cee9ee451e.png) ![image](https://user-images.githubusercontent.com/22197158/224308204-99c491b7-598a-40cd-bfb7-e65375cdd3b4.png)

I am using the same parameters that you are mentioning in the experiments section when you explain how you trained from scratch, using the pre-trained models from ImageNet and combined with the information provided in the multiple papers of Monodepth and Monodepth2.
![image](https://user-images.githubusercontent.com/22197158/224308689-d86ef938-7ea8-4412-94d8-5447c7169505.png)

When starting training the first mini-batch output is as shown in the following image:
![image](https://user-images.githubusercontent.com/22197158/224308617-e245ad6a-f5e7-4acc-b41c-d9b26f9f1f18.png)

However the following mini-batches from the same epoch are as follows, which means that the NN is not properly being trained:
![image](https://user-images.githubusercontent.com/22197158/224309115-b167a3b7-71a7-44cd-9199-49bdfd4ce77e.png) ![image](https://user-images.githubusercontent.com/22197158/224309155-b7d6274c-31a1-4119-9814-171c3ed2735f.png) ![image](https://user-images.githubusercontent.com/22197158/224309204-a03dc430-6a59-45c2-af77-78eb106ba134.png)

I am using the ZED2 camera which has a baseline of 12cm between each lenses and using rectified images before inference. 

With this provided information would you be able to understand what I may be missing from the papers and code provided that is not making the model train?

Thank you very much for your time.

**Comment by jordigc2** [comment]
If I reduce the Lr to 1e-4 and 5e-5 respectively from the ones mentioned in the screenshot of the paper the model does not saturate but when wrapping the images from left to right or vice-versa it gets this distortion on the edges.


Wrapped image from left-right
![image](https://user-images.githubusercontent.com/22197158/224316577-f8c18820-dead-48e5-9619-a5dfd394bb33.png)

Depth image after some mini-batches initial epoches:
![image](https://user-images.githubusercontent.com/22197158/224316410-061820be-12d0-42a7-815b-b7a8bf165a03.png)

After 14 epochs the depth image does not improve much from the previous one and it gets closer to a saturated depth image and the loss becomes constant.

Input and Estimated Depth:
![image](https://user-images.githubusercontent.com/22197158/224348933-f4ad44d7-2ffe-46fc-8afb-5bd0c97ede0d.png) ![image](https://user-images.githubusercontent.com/22197158/224348483-1ad38d9f-1509-4e11-a568-a4c44be22908.png)

Loss 14 epochs:
![image](https://user-images.githubusercontent.com/22197158/224348697-0a08b8ed-f7c7-4883-a4e1-f52bbf0c8694.png)



Thank you again for your time.

**Comment by zxcqlf** [comment]
So you mean you only use the left-right images for self-supervised training? not the sequential images? 
For the warped images, it is reasonable, because there are occlusions between left-right images.
For the depth, you know that self-supervised training is constructed on photometric consistency, those shadow regions affect the texture correspondence between images, resulting in wrong depth inferring results!

**Comment by jordigc2** [comment]
Hello,
I am also using the sequential images. I know that there are occlusions due to the stereo baseline, but it is only 12cm and the drift of the wrapped images is really big and never decreases when training from scratch. However, when using the pre-trained models you shared this drift are is smaller. 

So probably shadows are playing an important role in order to make the model learn? 
I will investigate how to mitigate the shadows noise before performing the photo-metric consistency. Are you aware of where can I start investigating this? Maybe that you may have considered looking at some point?

But the shadows between frames -1 and 1 compared with frame 0 do not change much, I do not understand how the shadows can affect the texture correspondence between images. 

Thank you very much.
Jordi
